flutter/flutter_config.go

```diff
@@ -47,7 +47,7 @@ func NewFromDirectory(dir string) (*FlutterConfig, error) {
 		return nil, err
 	} else if pubspec == nil {
 		// no pubspec found
-		return nil, nil
+		return nil, errors.New("no pubspec.yaml found in the current directory or any parent directory")
```

Comment:
Maybe extract this to `ErrNoPubspec`? No strong opinion though.

(And in tests you'd check that the correct error is thrown using `errors.Is`)